testplan/web_ui/testing/src/AssertionPane/AssertionTypes/basicAssertionUtils.js

```diff
@@ -153,6 +153,49 @@ function prepareRegexMatchLineContent(assertion) {
   return reconstructedString;
 }
 
+/*
+ * Prepare the contents of a report Attachment entry. Currently this just
+ * provides a link to download the raw file. In future we could inspect
+ * the filetype and e.g. render image files inline.
+ *
+ * @param {object} assertion
+ * @returns {object}
+ * @private
+ */
+function prepareAttachmentContent(assertion) {
+  const paths = window.location.pathname.split('/');
+  let downloadLink;
+
+  // When running the development server, the real Testplan back-end is not
+  // running so we can't GET the attachment. Stick in a button that
+  // gives a debug message instead of the real link.
+  if ((paths.length >= 2) && (paths[1] === '_dev')) {
+    downloadLink = (
+      <button onClick={() => alert("Would download: " + assertion.dst_path)}>
+        {assertion.filename}
+      </button>
+    );
+  } else if (paths.length >= 3) {
```

Comment:
Does the url when hosted on a server change as you view the testplan reports? I think in the dev mode hosted with npm, the url never changes from localhost:XXXX/_dev no matter where you browse in the UI. (or at least from what ive seen)

Reply:
yes, normally when you view a testplan report the URL is like `localhost:XXXX/testplan/[report UID]`, which we can inspect to extract the report UID. Running the development server and accessing ` localhost:XXXX/_dev` is a special case, which is why I have the branch above to check for that and render a button instead of a link.